src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/KubernetesCredentialsStore.java

```diff
@@ -0,0 +1,135 @@
+package com.cloudbees.jenkins.plugins.kubernetes_credentials_provider;
+
+import java.util.List;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import org.acegisecurity.Authentication;
+import org.apache.commons.lang.NotImplementedException;
+import org.jenkins.ui.icon.Icon;
+import org.jenkins.ui.icon.IconSet;
+import org.jenkins.ui.icon.IconType;
+import org.kohsuke.stapler.export.ExportedBean;
+import hudson.model.ModelObject;
+import hudson.security.ACL;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import com.cloudbees.plugins.credentials.Credentials;
+import com.cloudbees.plugins.credentials.CredentialsStore;
+import com.cloudbees.plugins.credentials.CredentialsStoreAction;
+import com.cloudbees.plugins.credentials.domains.Domain;
+
+public class KubernetesCredentialsStore extends CredentialsStore {
+
+    private final KubernetesCredentialProvider provider;
+    private final KubernetesCredentialsStoreAction action = new KubernetesCredentialsStoreAction(this);
+
+    public KubernetesCredentialsStore(KubernetesCredentialProvider provider) {
+        super(KubernetesCredentialProvider.class);
+        this.provider = provider;
+    }
+
+    @NonNull
+    @Override
+    public ModelObject getContext() {
+        return Jenkins.getInstance();
+    }
+
+    @Override
+    public boolean hasPermission(@NonNull Authentication authentication, @NonNull Permission permission) {
+        return Jenkins.getInstance().getACL().hasPermission(authentication, permission);
```

Comment:
As we don;t support Credential.ADD / UPDATE / REMOVE you should check these and return false (otherwise you would see the stack trace when you try to update / save)